src/nvim/ex_docmd.c

```diff
@@ -76,6 +76,17 @@
 #include "nvim/event/rstream.h"
 #include "nvim/event/wstream.h"
 
+/// The scope of a command.
+///
+/// The higher a number, the deeper a scope.
+enum cd_scope {
```

Comment:
Also modify `src/nvim/api/vim.c`. It should not call `post_chdir` with `false`, it should use one of the enum values.